components/patina_smbios/src/manager.rs

```diff
@@ -0,0 +1,2398 @@
+    pub fn new(record_type: u8) -> Self {
+        Self { record_type, data: Vec::new(), strings: Vec::new() }
+    }
+
+    /// Adds a field to the record
+    pub fn add_field<T>(mut self, value: T) -> Self
+    where
+        T: Copy + zerocopy::IntoBytes + zerocopy::Immutable,
+    {
+        self.data.extend_from_slice(value.as_bytes());
+        self
+    }
+
+    /// Adds a string to the record's string pool
+    pub fn add_string(mut self, s: String) -> Result<Self, SmbiosError> {
+        SmbiosManager::validate_string(&s)?;
+        self.strings.push(s);
+        Ok(self)
+    }
+
+    /// Builds the complete SMBIOS record
+    pub fn build(self) -> Result<Vec<u8>, SmbiosError> {
+        let mut record = Vec::new();
+
+        // Add header using zerocopy
+        let header = SmbiosTableHeader {
+            record_type: self.record_type,
+            length: (core::mem::size_of::<SmbiosTableHeader>() + self.data.len()) as u8,
+            handle: SMBIOS_HANDLE_PI_RESERVED,
+        };
+
+        record.extend_from_slice(header.as_bytes());
+
+        // Add data
+        record.extend_from_slice(&self.data);
+
+        // Add strings
+        if self.strings.is_empty() {
+            record.extend_from_slice(&[0, 0]);
+        } else {
+            for s in &self.strings {
+                record.extend_from_slice(s.as_bytes());
+                record.push(0);
+            }
+            record.push(0);
+        }
+
+        Ok(record)
+    }
+}
+
+/// Global storage for boot_services reference
+///
+/// This reference is stored during protocol installation and remains valid for
+/// the lifetime of the system. Required for TplMutex construction.
+///
+/// # Safety
+///
+/// - Initialized once during `install_smbios_protocol`
+/// - The boot_services reference must have 'static lifetime
+/// - Access is thread-safe due to UEFI's single-threaded DXE model
+struct GlobalBootServices {
+    boot_services: UnsafeCell<Option<&'static StandardBootServices>>,
+}
+
+unsafe impl Sync for GlobalBootServices {}
+
+impl GlobalBootServices {
+    const fn new() -> Self {
+        Self { boot_services: UnsafeCell::new(None) }
+    }
+
+    /// Initialize with a boot_services reference
+    ///
+    /// # Safety
+    ///
+    /// Must be called exactly once during system initialization.
+    /// The boot_services reference must have 'static lifetime.
+    unsafe fn initialize(&self, boot_services: &'static StandardBootServices) {
+        unsafe { *self.boot_services.get() = Some(boot_services) };
+    }
+
+    /// Get the stored boot_services reference
+    ///
+    /// # Safety
+    ///
+    /// Returns None if not initialized
+    #[allow(dead_code)] // Reserved for future diagnostic access to raw boot services
+    unsafe fn get(&self) -> Option<&'static StandardBootServices> {
+        unsafe { *self.boot_services.get() }
+    }
+}
+
+static BOOT_SERVICES: GlobalBootServices = GlobalBootServices::new();
```

Comment:
This should also be unecessary. You should be able to stash the BootServices you get from your component interface inside your service implementation.